### ISSUE TITLE
[Enhancement] Create functionality to manage GDG datasets 

### ISSUE THREAD

**Issue by destineddev** [ansible-collections/ibm_zos_core#935]
### Is there an existing issue for this?

- [X] There are no existing issues.

### Ansible module

zos_data_set

### Enhancement or feature description

As of today there is no way to access GDG data sets or create them. It would be great if there was a feature that would allow us to access a GDG dataset's generation values, delete, create new GDG datasets, and add a generation to an existing GDG. 
For example, `some.pdse.mem(latest) || some.pdse.mem(add) || some.pdse.mem(previous)`

**Comment by storfibm** [comment]
Efforts to provide support for GDG data sets has been committed to the underlying IBM Z Open Automation Utilities in upcoming releases. This will be a phased approach and tentatively `dcp`, `dtouch`, `dls` and `mvscmd` will receive this capability first as early as Q4. Subject to change. The ZOAU RFE can be tracked here: https://ibm-z-software-portal.ideas.ibm.com/ideas/ZOAU-I-10

For the z/OS Core Collection this means an implementation no sooner than sometime in 1H2024 tentatively. You may see other tickets that will get created around this capability.

**Comment by richp405** [comment]
Please follow updates in ticket 946:
https://github.com/ansible-collections/ibm_zos_core/issues/946

**Comment by richp405** [comment]
- [] #946 Related to adding GDG support

**Comment by ddimatos** [comment]
After reviewing all the updates, I recommend you follow this epic #1366 where GDG/GDS support will be tracked for Q2 2024. 

@destineddev please review #1366 to see if that meets your request. 

The only difference I see in issue #935 is that some keywords be supported, `latest`, `add`, `previous`, instead of `0`, `+1` , `-1`. While an intriguing idea, the difficulty in doing so is the code would spend considerable more time trying to determine if the data set `some.pdse.mem(latest)` belongs to a GDG (Generation Data Group) or is simply a PDS/E with member `latest`. 

In other words, the keywords support would introduce added runtime analysis reducing performance. Please review the issue #1366 to see if that meets your GDG support (without keywords). If keywords is still required, we would consider this after GDG is release to not complicate what is already changing 75% of the collections modules. Thus this would go into the backlog and be prioritized appropriately against critical items. 

Labeling this with "waiting for response" , it will close in 30 days without a response.
